openwisp_users/admin.py

```diff
@@ -409,7 +409,12 @@ def get_inline_instances(self, request, obj=None):
 
     def change_view(self, request, object_id, form_url='', extra_context=None):
         extra_context = extra_context or {}
-        obj = self.model.objects.get(pk=object_id)
+        try:
+            obj = self.model.objects.get(pk=object_id)
+        except self.model.DoesNotExist:
+            raise Http404("User Not Found.")
+        except ValidationError:
+            raise Http404("Invalid UUID")
         if user_not_allowed_to_change_owner(request.user, obj):
```

Comment:
To our ease, `django.contrib.admin.ModelAdmin` implements a `get_object` method that does error handling as well. Hence we can update the code as follows
```suggestion
        obj = self.get_object(request, object_id)
        if (obj is not None) and (user_not_allowed_to_change_owner(request.user, obj)):
```

Reply:
Made this change, works as expected now.